### PR TITLE
Enkhbilguun tserendorj

### PR DESCRIPTION
**Pull request by Bilguun1015**
worked the problems from 01 to 08.

### CODE REVIEW

File: src/01_bignum.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # Print out 2 to the 65536 power
 # (try doing the same thing in the JS console and see what it outputs)
 
-# YOUR CODE HERE
\ No newline at end of file
+# YOUR CODE HERE
+x = 2 ** 65536
+print(x)
```
Comment (brianilles) [inline src/01_bignum.py:6]: Nice! Love that you stored it in a variable

File: src/03_modules.py [hunk 2]
```diff
@@ -10,22 +10,24 @@
 
 # Print out the command line arguments in sys.argv, one per line:
 # YOUR CODE HERE
-
+for arg in sys.argv:
```
Comment (brianilles) [inline src/03_modules.py:13]: Awesome. sys.argv is just a list!

File: src/04_printing.py [hunk 3]
```diff
@@ -11,7 +11,8 @@
...
 # x is 10, y is 2.25, z is "I like turtles!"
-
+print('x is %d, y is %.2f, z is "%s"' %(x, y, z))
 # Use the 'format' string method to print the same thing
-
-# Finally, print the same thing using an f-string
\ No newline at end of file
+print('x is {}, y is {:0.2f}, z is "{}" '.format(x, y, z))
+# Finally, print the same thing using an f-string
+print(f'x is {x}, y ix {y:.2f}, z is "{z}"')
```
Comment (brianilles) [inline src/04_printing.py:18]: Good implementations! You can also use the escaping char `\` like so `print("x is {}, y is {:.2f}, z is \"{}\"".format(x, y, z))`

File: src/06_tuples.py [hunk 4]
```diff
@@ -37,8 +37,10 @@ def dist(a, b):
...
-print_tuple(t)  # Prints 1 2 5 7 99, one per line
-
+def print_tuple(t):  # Prints 1 2 5 7 99, one per line
+    for num in t:
+        print(num)
+print_tuple(t)
 # Declare a tuple of 1 element then print it
-u = (1)  # What needs to be added to make this work?
-print_tuple(u)
+u = (1,)  # What needs to be added to make this work?
```
Comment (brianilles) [inline src/06_tuples.py:45]: Good job figuring this out! Took me some time. The comma makes it a tuple

File: src/07_slices.py [hunk 5]
```diff
@@ -12,26 +12,27 @@
...
+middle = int(len(a)/2)
+print(a[middle-1], a[middle])
 
 # Output every element except the first one: [4, 1, 7, 9, 6]
-print()
+print(a[1:])
 
 # Output every element except the last one: [2, 4, 1, 7, 9]
-print()
+print(a[:-1])
```
Comment (brianilles) [inline src/07_slices.py:31]: Clean! You have the hang of this

File: src/08_comprehensions.py [hunk 6]
```diff
@@ -37,5 +40,7 @@
 
 # What do you need between the square brackets to make it work?
 y = []
-
+for num in x:
+    if (int(num) % 2) == 0:
+        y.append(num)
 print(y)
```
Comment (brianilles) [inline src/08_comprehensions.py:36]: Good. How would you write these as List comprehensions?

**Review by brianilles — APPROVED** [review]
Great start!

File: src/04_printing.py (hunk 3 above)
Comment (Bilguun1015) [reply src/04_printing.py:18]: Thank you.

**Review by Bilguun1015 — COMMENTED** [review]
changed all those for loops in 08 file into List comprehensions.

File: src/08_comprehensions.py [hunk 7]
```diff
@@ -39,8 +32,5 @@
 x = input("Enter comma-separated numbers: ").split(',')
 
 # What do you need between the square brackets to make it work?
-y = []
-for num in x:
-    if (int(num) % 2) == 0:
-        y.append(num)
+y = [ num for num in x if(int(num) % 2) == 0 ]
```
Comment (brianilles) [inline src/08_comprehensions.py:35]: Good work here! Note, in Python we actually don't need the parens for the conditional => `y = [num for num in x if int(i) % 2 == 0]` will suffice.

File: src/13_file_io.py [hunk 8]
```diff
@@ -9,10 +9,17 @@
 # Print all the contents of the file, then close the file
 
 # YOUR CODE HERE
-
+filename = "./foo.txt"
+file = open(filename, "r")
+for line in file:
+   print(line)
+file.close()
```
Comment (brianilles) [inline src/13_file_io.py:16]: Good job here. Another way to access a file is with `with open("file name here") as file: ...`. Cool thing is that we don't need to write `file.close()` with `with open()` as it is called automatically.

File: src/15_classes.py [hunk 9]
```diff
@@ -2,28 +2,41 @@
...
 # YOUR CODE HERE
-
+class Geocache(Waypoint):
+    def __init__(self, name, difficulty, size, lat, lon):
+        super().__init__(name, lat, lon)
+        self.difficulty = difficulty
+        self.size = size
+    def __str__(self):
+        return f'{self.name} diffculty level: {self.difficulty}, size: {self.size} at {self.lat} and {self.lon}'
 # Make a new waypoint and print it out: "Catacombs", 41.70505, -121.51521
```
Comment (brianilles) [inline src/15_classes.py:30]: Superb work with the basics of OOP in Python. How do you like this syntax compared with JS?

**Review by brianilles — APPROVED** [review]
Great work with day 1 and 2. It's clear you have built yourself up a solid base of understanding of Python. And, in such a short time, its quite impressive. Approving and closing this PR

**Pull request merged**